app/Http/Controllers/TestController.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Validator;
+use App\Test;
+use App\TestQuestion;
+use Illuminate\Http\Request;
+use App\Rules\ValidTestQuestionsScore;
+use App\Http\Resources\Test as TestAndQuestions;
+
+class TestController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $tests = Test::paginate(parent::PER_PAGE);
+
+        return TestAndQuestions::collection($tests);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        Validator::make($request->all(), [
+            'subject_id' => 'required|exists:subjects,id',
+            'group_id' => 'required|exists:groups,id',
+            'start' => 'required|date|after:now',
+            'end' => 'required|date|after:start',
+            'description' => 'string|max:65000|nullable',
+            'title' => 'string|max:255',
+            'questions' => [
+                'required', 'array', new ValidTestQuestionsScore
+            ],
+            'questions.*.score' => 'required|numeric',
```

Comment:
The score should be an integer value between 1 to 100.
With such validation, I can pass questions with negative scores and have a sum of scores in 100 ;)